src/map.rs

```diff
@@ -1348,6 +1348,38 @@ where
         self.put(key, value, false, guard)
     }
 
+    #[inline]
+    /// Tries to insert a key-value pair into the map.
+    ///
+    /// If the map did not have this key present, [`None`] is returned and the
+    /// key-value pair is inserted into the map.
+    ///
+    /// If the map did have this key present, and the old value is returned.
+    /// Neither the key or value are updated; this matters for types that can
+    /// be `==` without being identical. See the [std-collections
```

Comment:
I would maybe change `==` to refer to `Eq` here

Reply:
The same wording is used in the documentation of `flurry::HashMap::insert`, should I change it there too?

Reply:
Hm, this comes from https://github.com/jonhoo/flurry/commit/23a5fbae32d371da120512778959aa2e4e8de345. I still think `Eq` is clearer... @jonhoo opinion?

Reply:
I think the `==` comes from the `std` docs, but I agree that `Eq` (or really, `PartialEq`) is more explicit. That said, `==` **is** `PartialEq` in Rust, so :man_shrugging: I'm fine either way, but if @domenicquirl prefers `PartialEq`, then let's go with that. `Eq` I think is misleading.

Reply:
I do (even though I skipped over `PartialEq` as we require `Eq`). I just think it's more clear. Also, we should make it link to the corresponding `std` page.

Reply:
This isn't an issue anymore for us anymore, since we've altered the wording of the docs and `==` is no longer used. We should probably consider rephrasing `HashMap::insert` documentation though (which still uses `==`).

Reply:
> I do (even though I skipped over `PartialEq` as we require `Eq`). I just think it's more clear. Also, we should make it link to the corresponding `std` page.

Alright. `Eq` it is.

Reply:
Done!

Reply:
With the reword I still think this ends up going away?

Reply:
> With the reword I still think this ends up going away?

What do you mean? This now only affects the documentation `HashMap::insert`.

Reply:
Right, but if we're updating the docs of `HashMap::insert` to match the new language of `HashMap::try_insert`, then `Eq` would go away as a consideration there as well.

Reply:
I think this has been resolved, hasn't it?

Reply:
> Right, but if we're updating the docs of `HashMap::insert` to match the new language of `HashMap::try_insert`, then `Eq` would go away as a consideration there as well.

I see, I'll have a look at it.